src/app/services/events.tsx

```diff
@@ -223,3 +229,37 @@ export const userCanReserveEventSpaces = (user: Immutable<PotentialUser> | null,
     isTeacherOrAbove(user)
   );
 };
+
+export function googleCalendarTemplate(event: CurrentEventState | undefined) {
```

Comment:
Since the only usage for this function is an `AugmentedEvent`, why not make that the required type? That way this function becomes simpler, as you don't need to care about undefined or null methods.

Reply:
I hadn't messed with this function (just changed where it lived!) and I presume they set it up based on the fact that `event` in the EventDetails component could be CurrentEventState (which is Augmented Event, 404 or null) or undefined, however because it's inside a "Show Loading" component, it essentially guarantees that the google calendar button wouldn't render unless there was an Augmented Event. Changed it now!